src/mods.cpp

```diff
@@ -28,11 +28,44 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "util/strfnd.h"
 #include "convert_json.h"
 #include "exceptions.h"
+#include <vector>
+
+static void independentGetLine(std::istream& is, std::string& line)
+{
+    // The characters in the stream are read one-by-one using a std::streambuf.
```

Comment:
I think the best approach would be to just search for `\n` and split after that, removing all leftover `\r` characters in the string. I think the function `str_replace` from `util/string.h` can do this.

There are no modern OSs which still use `\r` so this should be safe to do, and we can avoid adding such a big function to the codebase (whose origin isn't clear as well copyright wise).